localstack-core/localstack/aws/handlers/openapi.py

```diff
@@ -0,0 +1,71 @@
+"""
+Handlers for validating request and response schema against OpenAPI specs.
+"""
+
+import logging
+
+from openapi_core import OpenAPI
+from openapi_core.contrib.werkzeug import WerkzeugOpenAPIRequest, WerkzeugOpenAPIResponse
+from openapi_core.exceptions import OpenAPIError
+from openapi_core.validation.request.exceptions import (
+    RequestValidationError,
+)
+
+from localstack import config, spec
+from localstack.aws.api import RequestContext
+from localstack.aws.chain import Handler, HandlerChain
+from localstack.constants import INTERNAL_RESOURCE_PATH
+from localstack.http import Response
+
+LOG = logging.getLogger(__name__)
+
+
+class OpenAPIRequestValidator(Handler):
+    """
+    Validates the internal requests with the OpenAPI spec.
+    """
+
+    def __init__(self):
+        self.openapi = OpenAPI.from_dict(spec.OPENAPI)
+
+    def __call__(self, chain: HandlerChain, context: RequestContext, response: Response):
+        if not config.OPENAPI_VALIDATE_REQUEST:
+            return
+
+        path = context.request.path
+
+        if path.startswith(f"{INTERNAL_RESOURCE_PATH}/") or path.startswith("/_aws/"):
+            try:
+                self.openapi.validate_request(WerkzeugOpenAPIRequest(context.request))
+            except OpenAPIError as e:
+                # Note: we only check request body validation errors, like invalid body requests or missing required
+                #   parameters. Other things we do currently explicitly check are ServerNotFound, OperationNotFound,
+                #   and PathNotFound. They are most likely intercepted in the handler chain.
```

Comment:
I am a bit confused by this comment:

> "Other things we do currently explicitly check are ServerNotFound, OperationNotFound, and PathNotFound."

I guess that should mean "Other things that we currently explicitly do _not_ check"?

> They are most likely intercepted in the handler chain.

I am a bit confused by this sentence. What does it mean / tell us?